guide: make plots guide

dberenbaum opened the pull request:
This is a quick iteration to address https://github.com/iterative/dvc.org/issues/3818. I made a new guide page since I thought shoving that all into the dvcyaml/project structure would be way too much/messy and hard to find.

Also related: https://github.com/iterative/dvc.org/issues/2572

In review app: https://dvc-org-guide-plots-u9q3eipvdr.herokuapp.com/doc/user-guide/plots

jorgeorpinel commented:
Note: this may be closer to addressing (part of) https://github.com/iterative/dvc.org/issues/2572 than https://github.com/iterative/dvc.org/issues/3818

jorgeorpinel commented on content/docs/sidebar.json:
```diff
@@ -154,6 +154,10 @@
         ]
       },
+      {
+        "label": "Plots Configuration and Visualization",
+        "slug": "plots"
```
Name's a bit long. It'd be best if it fits in one line in the nav.

<img width="399" alt="image" src="https://user-images.githubusercontent.com/1477535/182013944-eec20bfd-2842-4fff-b8ec-cb69cadb9dcd.png">

Also that the slug marches the title if possible (better for SEO AFAIK).

dberenbaum replied:
Agreed. Do you think `Plots` is enough? What do you suggest?

jorgeorpinel replied:
Maybe "Data Visualization" for now?

Or something else that will let us include params and metrics (in general) in the future too, per https://github.com/iterative/dvc.org/issues/2572 but we can change it later.

dberenbaum replied:
> Maybe "Data Visualization" for now?

I like to at least use the word "Plots" or it feels like needless obfuscation. Do you think so, or is "Data Visualization" more accurate in some way?

> Or something else that will let us include params and metrics (in general) in the future too, per https://github.com/iterative/dvc.org/issues/2572 but we can change it later.

I feel plots may need its own guide since it's a fairly complex subsystem of DVC on its own, but yeah, I guess it's premature to decide either way right now.

jorgeorpinel replied:
Maybe this for now
```suggestion
        "slug": "visualizing-plots"
```
Up to you, I just think just "Plots" is too generic. But we expect guide traffic to come from links in other docs (rather than by browsing the nav) so the name may not be that important.

dberenbaum replied:
`Visualizing plots` works and fits the style of the rest of the other guides, so I'll go with that.

This is mostly a style choice, but I prefer to keep it simple because IMHO it's easier to navigate if the sidebar looks like:

* Experiments
* Pipelines
* Plots

It takes longer for me to find the right section when we add in these descriptive words like:

* Experiments management
* Data pipelines
* Visualizing plots

jorgeorpinel replied:
Goes back to how exactly we want to organize the user guide. I don't think we can maintain detailed guides for every single feature so specific titles may be more appropriate. Up to you

jorgeorpinel commented on content/docs/command-reference/plots/index.md:
```diff
@@ -25,311 +25,11 @@ positional arguments:
-for example. Plots are a great alternative to `dvc metrics` when working with
-multi-dimensional performance data. They also help you present and compare
-[experiments] effectively.
+`dvc plots` subcommands configure and visualize data and images produced by
+machine learning projects.
```
I see this follows the pattern in the [`exp` ref](https://dvc.org/doc/command-reference/exp#description). That can work 👍🏼 

But what about going with something like the [`cache`](https://dvc.org/doc/command-reference/cache) and  [`stage`](https://dvc.org/doc/command-reference/stage#description) refs instead? That is: a brief paragraph around the concept in question, then one with a summary of each subcommand.

I'm honestly not sure which pattern to go with but it may be a good time to decide (we'll need it again to finish https://github.com/iterative/dvc.org/issues/2572).

For now, quick suggestion on existing text:
```suggestion
`dvc plots` subcommands help you visualize and compare data or images
produced by machine learning projects, letting you fully customize graph
settings and style.
```

jorgeorpinel commented on content/docs/command-reference/plots/index.md:
```diff
@@ -338,116 +38,3 @@ this feature to render DVC plots without an Internet connection, below.
 
 - `-v`, `--verbose` - displays detailed tracing information.
-
-## Example: Offline HTML Template
-
-The plots generated by `dvc plots` uses Vega-Lite JavaScript libraries, and by
-default these load [online resources](https://vega.github.io/vega/usage/#embed).
-There may be times when you need to produce plots without Internet access, or
```
I think all these examples still belong more to the cmf ref, at least for consistency (open to discuss) ...

jorgeorpinel replied:
> Would you suggest putting them in https://dvc.org/doc/command-reference/plots/show?

No @dberenbaum I was thinking to keep them where they were but I realize now that's out of context since all the plot template info is in the new guide now. But...

jorgeorpinel commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+[example](/doc/user-guide/plots#example-offline-html-template) that uses this
+feature to render DVC plots without an Internet connection, below.
+
+## Example: Offline HTML Template
+
+The plots generated by `dvc plots` uses Vega-Lite JavaScript libraries, and by
+default these load [online resources](https://vega.github.io/vega/usage/#embed).
+There may be times when you need to produce plots without Internet access, or
```
... In the user guide we don't usually have separate examples like these. We embed them into the explanations instead. Think book format.

jorgeorpinel replied:
> all the plot template info is in the new guide now

... ideally we would dissolve the examples into the explanations themselves then, as suggested here.

> I'm trying to reorganize first as a quick iteration

OK. Let's try to check we have a check box somewhere to follow up on this.

jorgeorpinel commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+# Plots Configuration and Visualization
+
```
> I made a new guide page since I thought shoving that all into the dvcyaml/project structure would be way too much

I definitely agree to put the explanations in this new page! It's a great step towards https://github.com/iterative/dvc.org/issues/2572. But ...

jorgeorpinel commented on content/docs/user-guide/plots.md:
````diff
@@ -0,0 +1,420 @@
+}
+```
+
+## Defining plots
+
+In order to create visualizations, users need to provide the data and
+(optionally) configuration that will help customize the plot. DVC provides two
````
... what I'm suggesting to summarize (heavily probably) and move to the dvc.yaml spec file is everything in this section. It's mainly about the schema.

jorgeorpinel replied:
> use the guide to show how to setup plots and leave the cmd ref and dvc.yaml pages for lookup purposes

jorgeorpinel replied:
> 2... use the guide to show how to setup plots and leave the cmd ref and dvc.yaml pages for lookup purposes

We agree on this @dberenbaum. I just think the [Top-level plots](https://dvc-org-guide-plots-u9q3eipvdr.herokuapp.com/doc/user-guide/plots#top-level-plots) explanation can be shorter and leave most schema details [for the dvc.yaml "guide"](https://dvc-org-guide-plots-u9q3eipvdr.herokuapp.com/doc/user-guide/project-structure/dvcyaml-files#top-level-plot-definitions).

E.g. phrases like "should be defined ... under plots key" or "Configuration ... should be a dictionary", and the whole **Available configuration fields** section (as noted). In short, this section is currently a more complete ref than the actual ref.

jorgeorpinel commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+  https://github.com/iterative/vscode-dvc/blob/main/extension/resources/walkthrough/plots.md
+[experiments]: /doc/user-guide/experiment-management/experiments-overview
+
+### Supported file formats
```
```suggestion
## Supported file formats
```

jorgeorpinel commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+Note that in the case of CSV/TSV metrics files, column names from the table
+header (first row) are equivalent to field names.
+
+### Custom templates
```
> Unrelated but should we rename this **Custom Vega templates** since there's a **Custom HTML templates** H2 right after this?

pared commented on content/docs/command-reference/plots/diff.md:
```diff
@@ -147,9 +146,8 @@ cat,turtle
-The predefined confusion matrix
-[template](/doc/command-reference/plots#plot-templates) (in
-`.dvc/plots/confusion.json`) shows how metrics comparisons can be faceted by
+The predefined confusion matrix [template](/doc/user-guide/plots#plot-templates)
```
```suggestion
The predefined confusion matrix [template](/doc/user-guide/plots#plot-templates-data-series-only)
```

pared commented on content/docs/command-reference/plots/diff.md:
```diff
@@ -67,13 +66,13 @@ all the current plots, without comparisons.
-  [plot template](/doc/command-reference/plots#plot-templates) to be injected
-  with data. The default template is `.dvc/plots/default.json`. See more details
-  in `dvc plots`.
+  [plot template](/doc/user-guide/plots#plot-templates) to be injected with
```
```suggestion
[plot template](/doc/user-guide/plots#plot-templates-data-series-only) to be injected with
```

pared commented on content/docs/command-reference/plots/diff.md:
```diff
@@ -39,9 +39,8 @@ All plots defined in `dvc.yaml` are used by default, but specific files can be
-[plot templates](/doc/command-reference/plots#plot-templates), using the
-`--template` option. To learn more about plots files and templates please see
-`dvc plots`.
+[plot templates](/doc/user-guide/plots#plot-templates), using the `--template`
```
```suggestion
[plot templates](/doc/user-guide/plots#plot-templates-data-series-only), using the `--template`
```

pared commented on content/docs/command-reference/plots/modify.md:
```diff
@@ -44,7 +44,7 @@ Note that a secondary use of this command is to convert output or simple
 
 - `-t <name_or_path>, --template <name_or_path>` - set a default
-  [plot template](/doc/command-reference/plots#plot-templates).
+  [plot template](/doc/user-guide/plots#plot-templates).
```
```suggestion
[plot template](/doc/user-guide/plots#plot-templates-data-series-only).
```

pared commented on content/docs/command-reference/plots/show.md:
```diff
@@ -39,9 +39,9 @@ The default behavior of this command can be modified per [stage plot] file with
-[plot templates]: /doc/command-reference/plots#plot-templates
-[top-level plot]: /doc/command-reference/plots#top-level-plots
-[stage plot]: /doc/command-reference/plots#stage-plots
+[plot templates]: /doc/user-guide/plots#plot-templates
```
```suggestion
[plot templates]: /doc/user-guide/plots#plot-templates-data-series-only
```

pared commented on content/docs/command-reference/plots/show.md:
```diff
@@ -50,13 +50,13 @@ The default behavior of this command can be modified per [stage plot] file with
-  [plot template](/doc/command-reference/plots#plot-templates) to be injected
-  with data. The default template is `.dvc/plots/default.json`. See more details
-  in `dvc plots`.
+  [plot template](/doc/user-guide/plots#plot-templates) to be injected with
```
```suggestion
  [plot template](/doc/user-guide/plots#plot-templates-data-series-only) to be injected with
```

pared commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+
+In that case the default behavior will be applied. DVC will take data from
+`logs.csv` file and apply `linear` plot
+[template](/doc/user-guide/plots#plot-templates) to the last found column (CSV,
```
```suggestion
[template](/doc/user-guide/plots#plot-templates-data-series-only) to the last found column (CSV,
```

pared commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+
+The `linear` template is the default. It can be changed with the `--template`
+(`-t`) option of `dvc plots show` and `dvc plots diff`. The argument provided to
+`--template` can be a (built-in) template name or a path to a [custom template].
```
```suggestion
`--template` can be a (built-in) template name or a path to a [custom template](#custom-templates).
```

pared commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+- `confusion` - confusion matrix, see
+  [example](/doc/user-guide/plots#example-confusion-matrix)
+
+[custom templates]: https://dvc.org/doc/command-reference/plots/templates
```
```suggestion
[custom templates]: #custom-templates
```

pared reviewed (commented):
Seems like I left a lot of wrong links on #3691 and driven by passing link reports didn't check it thoroughly. Left some suggestions.
Besides that LGTM.

jorgeorpinel commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,420 @@
+
+- `linear` - basic linear plot including cursor interactivity (default)
+- `simple` - simplest linear template (not interactive); Good base to create
+  [custom templates].
```
To complement https://github.com/iterative/dvc.org/pull/3830#pullrequestreview-1057430160
```suggestion
  custom templates.
```

jorgeorpinel commented:
> a lot of wrong links on https://github.com/iterative/dvc.org/pull/3691 and driven by passing link reports didn't check it thoroughly

Thanks @pared , good catches. I committed them all.

The link checker is unable to check the anchors in the link, it only checks that the HTTP return code is 400.

dberenbaum commented:
Thanks @jorgeorpinel! Details below, but to summarize my thoughts:
1. What do you think of limiting this PR to reorganizing the current content and leaving any content changes to a future PR?
2. I'd like to use the guide to show how to setup plots and leave the cmd ref and dvc.yaml pages for lookup purposes.

> I think all these examples still belong more to the cmf ref, at least for consistency (open to discuss) ...

Would you suggest putting them in https://dvc.org/doc/command-reference/plots/show? Already, most examples there don't show any new command uses or options but instead show different ways to configure plots (most of the examples are simply `dvc plots show` or at most add a target).

> ... In the user guide we don't usually have separate examples like these. We embed them into the explanations instead. Think book format.

Makes sense. I'm trying to reorganize first as a quick iteration and then we can work on improving the content. What do you think?

> ... what I'm suggesting to summarize (heavily probably) and move to the dvc.yaml spec file is everything in this section. It's mainly about the schema.

Summarizing makes sense. Again, I'm trying to reorganize first.

I agree that the `Available configuration fields` at least could be moved.

My feeling is that users need explanation of how to configure plots in dvc.yaml and not just a technical reference. Users who want to learn how to configure a plot can find a top-level plots guide easier than looking into `Project structure` -> `dvc.yaml files`. I also think that splitting information between the cmd ref, dvc.yaml guide, and plots guide makes it harder for users.

jorgeorpinel reviewed (approved):
> limiting this PR to reorganizing the current content and leaving any content changes to a future PR?

Sure, approving

> users need explanation of how to configure plots in dvc.yaml and not just a technical reference...

"explanation" and "how-to" are unclear terms rn 😅 but yes, a plots guide including definition and setup options makes sense.

> Users who want to learn how to configure a plot can find a top-level plots guide easier

For initial learning there's https://dvc.org/doc/start/data-management/metrics-parameters-plots though.

> easier than looking into Project structure -> dvc.yaml files

p.s. maybe we should make a /doc/format-reference section for these "guides" that are actually references.

jorgeorpinel commented:
> splitting information between the cmd ref, dvc.yaml guide, and plots guide makes it harder for users

And Get Started... This is why we need a clear docs framework ⏳

dberenbaum commented:
> Sure, approving

Thanks @jorgeorpinel! I didn't mean to pressure you into approval. Since the content already exists in the cmd ref, let's continue to iterate more on the structure here rather than rush to merge it. I just don't want to also add in new content or I think it will get too unwieldy.

> > Users who want to learn how to configure a plot can find a top-level plots guide easier
> 
> For initial learning there's https://dvc.org/doc/start/data-management/metrics-parameters-plots though.
> 
> > easier than looking into Project structure -> dvc.yaml files
> 
> p.s. maybe we should make a /doc/format-reference section for these "guides" that are actually references.

I don't see it as much different from https://github.com/iterative/dvc.org/pull/3414.

I imagined the docs structure to be:
1. Get started gives an intro to features like plots or pipelines.
2. User guide gives a comprehensive explanation of the features for users who want to go deeper.
3. Cmd ref provides an organized way to quickly lookup needed info for experienced users.

Does that match your mental model?

By the way, I plan to make changes to the PR but want to wait until we have some consensus on how it should be organized before addressing any comments so far.

jorgeorpinel commented:
> 2. User guide gives a comprehensive explanation of the features for users who want to go deeper...
>
> Does that match your mental model?

Except on 2. Like I said, some words like "explanation" or "guide" are unclear rn. Anything can match that description really... but anyway. In my currently [proposed framework](https://www.notion.so/iterative/Framework-7e86e484e50e42529df3dbfe32b4e987), there are (high-level) tutorial-type docs and (deeply technical) explanation-type docs (among others). I'm not sure which one you're intending here, but ideally you shouldn't have to worry about producing the latter (drafts or structure plans could be super useful though).

One of my goals for this framework is to keep much of the /doc/user-guide technical, so the product teams can mostly own those parts. So maybe the question is what would you be comfortable writing and maintaining?

shcheklein commented:
Only thing to add to this. I would try to avoid making command references way too short. They should be like a man page still. It has some dry formal description of a behavior, some examples to grasp the idea fast, etc.

User guide is definitely higher level. In case of plots, similar to the current index page https://dvc.org/doc/command-reference/plots I think. And can go even a bit higher level to describe things like how to get the data in the first place - `dvclive`, dump it with Python. Can describe the differences between this and usual loggers, etc, etc.

jorgeorpinel commented:
> avoid making command references way too short. They should be like a man page

Agree

> User guide is definitely higher level

Hmmm. Yeah TBH the UG is the most confusing part of our docs conceptually rn, probably because it is both high level and very low-level some times (e.g. it includes implementation details). Maybe there's no way to completely separate them to have a true book-like "guide". But also the level of effort to keep a complete UG like that (which may overlap a lot with other types of docs) is pretty high -- something to consider.

dberenbaum commented:
> User guide is definitely higher level. In case of plots, similar to the current index page https://dvc.org/doc/command-reference/plots I think.

To clarify, that's what this PR currently does.

However, I understand the concern from @jorgeorpinel that what's there is too low level and more of a reference than a guide. 🤔 ⌛

dberenbaum commented on content/docs/user-guide/plots.md:
```diff
@@ -0,0 +1,421 @@
+`test_logs.csv` against the `epoch`. Note that both files have to have `epoch`
+field.
+
+### Available configuration fields
```
@pared Should `template` be added here?

dberenbaum commented:
> To clarify, that's what this PR currently does.
> 
> However, I understand the concern from @jorgeorpinel that what's there is too low level and more of a reference than a guide. 🤔 ⌛

I did another pass and made some changes:

* Moved a lot of the top-level plots configuration and available configuration fields to the dvcyaml guide.
* Moved custom template configuration to the plots templates cmd ref.
* Moved most examples to the plots show cmd ref.
* Duplicated a few high-level examples in the plots user guide.

shcheklein commented on content/docs/user-guide/visualizing-plots.md:
```diff
@@ -0,0 +1,334 @@
+# Visualizing Plots
```
It's still low level to my mind. From reading this it's not clear (at least not immediately) how is different from just any plot lib. It reads like it's an alternative to that, while in our case the core differentiator is the approach - codification. DVC captures data into commits and then it can be extracted in different ways to do stuff (including plots). I think it's important to describe.

+ (or along the way) I would give a brief sense how it works - files are saved (can be written from a script in any format) + then DVC provides templates, tools etc to visualize this.

dberenbaum replied:
> What do you think of limiting this PR to reorganizing the current content and leaving any content changes to a future PR?

@shcheklein Are you okay with this approach? Wanted to focus on where to put the existing content first.

shcheklein replied:
If we prioritize and do this quick - it makes sense. Otherwise I would be neutral about this PR.

Worst case - we move it now, next someone decides to move things other direction, and so on ... we'll be just moving things around. That's what we need to avoid at all cost.

jorgeorpinel commented on content/docs/sidebar.json:
```diff
@@ -154,6 +154,9 @@
           "share-a-dvc-cache"
         ]
       },
+      {
+        "slug": "visualizing-plots"
+      },
```
This can be shortened to just

```suggestion
      "visualizing-plots"
```

p.s. there's an unnecessary new line in line 167.

jorgeorpinel replied:
UPDATE: I'm fixing these in https://github.com/iterative/dvc.org/pull/3903

jorgeorpinel replied:
> I already submitted a PR in https://github.com/iterative/dvc.org/pull/3902

Oops. My bad

jorgeorpinel commented on content/docs/user-guide/visualizing-plots.md:
```diff
@@ -0,0 +1,385 @@
+
+## Generating plots files
+
+To generate the data files for plots, you can:
+
+1. Use [DVCLive](/doc/dvclive/dvclive-with-dvc) in your Python code to log the
+   data in the expected format for you.
+2. Save data yourself in one of the
```
💅🏼 Maybe a regular bullet list (not numbered) would be better here to signal equal alternatives and contrast to previous numbered list which was a list of steps.

jorgeorpinel commented on content/docs/user-guide/visualizing-plots.md:
````diff
@@ -0,0 +1,385 @@
+    y: tpr
+```
+
+3. [Show](/doc/command-reference/plots/show) all plots in a single view or
+   report.
````
```suggestion
3. `dvc plots show` all plots in a single view or report.
```

dberenbaum replied:
Keeping this one as is. Don't want to introduce commands here since it's too low level and inconsistent with the others, plus it's inaccurate for showing plots in VS Code and Studio.

jorgeorpinel commented on content/docs/user-guide/visualizing-plots.md:
```diff
@@ -0,0 +1,385 @@
+![](/img/plots_cm_get_started_show.svg)
+
+4. Run [experiments](/doc/user-guide/experiment-management/experiments-overview)
+   and [compare](/doc/command-reference/plots/diff) plots.
```
```suggestion
   and [compare](#comparing-plots) plots.
```

jorgeorpinel commented on content/docs/user-guide/project-structure/dvcyaml-files.md:
```diff
@@ -477,34 +475,98 @@ validation and auto-completion.
-The list of `plots` contains one or more user-defined `dvc plots` (paths
+The list of `plots` contains one or more user-defined top-level plots (paths
 relative to the location of `dvc.yaml`).
 
-This example makes output `auc.json` viable for visualization, configuring keys
-`fpr` and `tpr` as X and Y axis, respectively:
+Every plot has to have its own ID. Configuration, if provided, should be a
+dictionary.
```
Combine intro P
```
The list of `plots` contains one or more user-defined top-level plots (paths
relative to the location of `dvc.yaml`).
Every plot needs a unique file path as identifier.
Optional configuration can be given as a dictionary.
```

dberenbaum replied:
> Every plot needs a unique file path as identifier.

You can have many different plots all based on the same file.

jorgeorpinel replied:
> different plots all based on the same file

I see. This "path vs ID" situation is a bit confusing. Explanations could improve on that I think... ⌛

jorgeorpinel commented on content/docs/user-guide/project-structure/dvcyaml-files.md:
````diff
@@ -477,34 +475,98 @@ validation and auto-completion.
+dictionary.
+
+In the simplest use case, a user can provide the file path as the plot ID and
+not provide configuration at all:
 
 ```yaml
-stages:
-  build:
````
Consolidate Ps about this example

```
In the simplest use, you can provide the plot's file path and no configuration.
In that case, the default behavior will be applied.
In the example below, DVC will take data from
`logs.csv` and apply the `linear` plot
[template](/doc/user-guide/visualizing-plots#plot-templates-data-series-only) to
the last found column:
```

> Remove redundant P after example.

jorgeorpinel commented on content/docs/user-guide/project-structure/dvcyaml-files.md:
````diff
@@ -477,34 +475,98 @@ validation and auto-completion.
+[template](/doc/user-guide/visualizing-plots#plot-templates-data-series-only) to
+the last found column (CSV, TSV files) or field (JSON, YAML).
+
+We can customize the plot by adding appropriate fields to the configuration:
+
+```yaml
+# dvc.yaml
````
Ideally follow same pattern (1 paragraph per example).

jorgeorpinel commented on content/docs/user-guide/project-structure/dvcyaml-files.md:
````diff
@@ -477,34 +475,98 @@ validation and auto-completion.
+`x` axis we will have `actual_class` field. Note that DVC will assume that
+`actual_class` is inside `confusion_matrix_data.csv`.
+
+We can provide multiple columns/fields from the same file:
+
+```yaml
````
Same

jorgeorpinel commented on content/docs/user-guide/project-structure/dvcyaml-files.md:
````diff
@@ -477,34 +475,98 @@ validation and auto-completion.
+In this case, we will take `accuracy` and `loss` fields and display them agains
+`epoch` column, all coming from `logs.csv` file.
+
+We can source the data from multiple files too:
+
+```yaml
````
=

jorgeorpinel commented:
Thanks for all the updates and sorry for late check @dberenbaum . Left some specific suggestions (minor) above...

https://dvc.org/doc/user-guide/visualizing-plots and https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#top-level-plot-definitions did end up having all the info they need I think. Of course we can always improve on docs but at least they seem to serve their intended purpose as they are now 👍🏼 

Some thoughts:

- "Save data to a supported file format": Should we emphasize that the data is produced by a data pipeline? Link to `#generating-plots-files` maybe.
- "Show all plots in a single view": Can the figures be consolidated into a single smaller image? Lots of scrolling rn.
- We can combine the "Generating plots files" and "Supported file formats" sections, I think. Maybe "Plot data sources" or some other title.
- Move "Top-level plot" section before "Stage plots" ?
  And link to https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#top-level-plot-definitions directly.
- dvc.yaml doc ([Top-level plot definitions](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#top-level-plot-definitions))

jorgeorpinel reviewed (commented):
Relatively minor review of dvc.yaml doc changes (for readability)
